vta/tutorials/frontend/deploy_detection_on_vta.py

```diff
@@ -0,0 +1,334 @@
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+Deploy Pretrained Vision Detection Model from Darknet on VTA
+================================================
+**Author**: `Hua Jiang <https://github.com/huajsj>`_
+
+This tutorial provides an end-to-end demo, on how to run Darknet YoloV3-tiny
+inference onto the VTA accelerator design to perform Image detection tasks.
+It showcases Relay as a front end compiler that can perform quantization (VTA
+only supports int8/32 inference) as well as graph packing (in order to enable
+tensorization in the core) to massage the compute graph for the hardware target.
+"""
+
+######################################################################
+# Install dependencies
+# --------------------
+# To use the autotvm package in tvm, we need to install some extra dependencies.
+# (change "3" to "2" if you use python2):
+#
+# .. code-block:: bash
+#
+# pip3 install "Pillow<7"
+#
+# YOLO-V3-tiny Model with Darknet parsing have dependancy with CFFI and CV2 library,
+# we need to install CFFI and CV2 before executing this script.
+# 
+# pip3 install "Pillow<7"
+#
+# pip3 install cffi
+# pip3 install opencv-python
+#
+# Now return to the python code. Import packages.
+
+from __future__ import absolute_import, print_function
+
+import sys
+import os
+import time
+import tvm
+import vta
+import numpy as np
+import matplotlib.pyplot as plt
+import tvm.relay.testing.yolo_detection
+import tvm.relay.testing.darknet
+
+from ctypes import *
+from tvm import relay
+from tvm.contrib import graph_runtime
+from tvm.contrib.download import download_testdata
+from tvm.contrib import graph_runtime, util
+from tvm.relay.testing.darknet import __darknetffi__
+
+
+from vta.testing import simulator
+from vta.top import graph_pack
+from tvm import rpc, autotvm, relay
+# Make sure that TVM was compiled with RPC=1
+assert tvm.runtime.enabled("rpc")
+
+######################################################################
+# Download yolo net configure file, weight file, darknet library file based on
+# Model Name
+# --------------------------
+MODEL_NAME = 'yolov3-tiny'
+REPO_URL = 'https://github.com/dmlc/web-data/blob/master/darknet/'
+
+cfg_path = download_testdata('https://github.com/pjreddie/darknet/blob/master/cfg/' 
+                             + MODEL_NAME + '.cfg' + '?raw=true', 
+                             MODEL_NAME + '.cfg', 
+                             module="darknet")
+weights_path = download_testdata('https://pjreddie.com/media/files/' 
+                                 + MODEL_NAME + '.weights' + '?raw=true', 
+                                 MODEL_NAME + '.weights', 
+                                 module="darknet")
+
+if sys.platform in ['linux', 'linux2']:
+    darknet_lib_path = download_testdata(REPO_URL + 'lib/' + 'libdarknet2.0.so' + '?raw=true', 
```

Comment:
the reason to separate the string is to try to make the "download" more readable, if we concat the 'libdarknet2.0.so' with lib and '?raw=true', it would become difficult to get the information about what the download try to do.